addons/api/addon/serializers/application.js

```diff
@@ -39,6 +39,8 @@ export default class ApplicationSerializer extends RESTSerializer {
   serializeAttribute(snapshot, json, key, attribute) {
     const { type, options } = attribute;
     let value = super.serializeAttribute(...arguments);
+    // Remove secret attributes that are null or empty
+    if (options.isSecret && !json[key]) delete json[key];
```

Comment:
Since this is in the application serializer, should we be more specific than checking a falsy value? It's unlikely but in the future, I wouldn't want to get caught accidentally stripping out something like `0`.

Reply:
Yes, I can add a test case for that.

Reply:
I updated this to check the type. We will add another type check once JSON creds are implemented so this would have been needed.